packages/client/src/utils.js

```diff
@@ -46,39 +46,47 @@ export const isAddr = (addr) => {
 export const formatAddress = (addr) => {
   return addr.startsWith("0x") ? addr : `0x${addr}`;
 };
-export const formatActionString = (str) => {
-  const isFungibleTransfer = str.includes("FungibleToken.Receiver");
-  let newStr = "";
-  const words = str.split(" ");
-  words.forEach((word, idx) => {
-    const noLength = !word?.trim().length;
-    //remove class names like Capability<&AnyResource{A.0x.FungibleToken.Receiver}>
-    const isClass = word.startsWith("Capability<");
-    if (noLength || isClass) {
-      return;
-    }
-    const float = parseFloat(word);
-    if (isNaN(float) || word.startsWith("0x")) {
-      // add FLOW to token transfers
-      if (isFungibleTransfer && word === "tokens") {
-        newStr += "FLOW ";
-      }
-      newStr += word;
-    } else {
-      newStr += float;
-    }
-    // add space if not done
-    if (idx < words.length - 1) {
-      newStr += " ";
-    }
-  });
-
-  // remove "from the treasury" as thats implied
-  newStr = newStr.replace("from the treasury ", "");
 
-  return newStr;
+export const formatActionString = (str) => {
+  const vaultRegex = /A\..*\.Vault/g;
+  const collectionRegex = /A\..*\.Collection/g;
+  const floatRegex = /[0-9]*[.][0-9]+/;
+  const isTokenTransfer = vaultRegex.test(str);
+  const isNFTTransfer = collectionRegex.test(str);
+  let contractName;
+  let result;
+  if (isTokenTransfer) {
+    contractName = str.match(vaultRegex)[0].split(".")[2];
+    const tokenName = getTokenName(contractName);
+      result = str.replace(floatRegex, parseFloat(str.match(floatRegex)[0]));
+    return result.replace(vaultRegex, tokenName);
+  }
+  if (isNFTTransfer) {
+    contractName = str.match(collectionRegex)[0].split(".")[2];
+    return str.replace(collectionRegex, contractName);
+  }
+  return str;
 };
 
+export const getTokenName = (contractName) => {
+  switch (contractName) {
```

Comment:
Can we utilize COIN_TYPE_TO_META to get the tokenName instead? 
similar as this: https://github.com/DapperCollectives/VESSEL/blob/ba950e458a42315c4670a331be8f2dfeea2eeff7/packages/client/src/components/SendTokens/components/CoinTypeDropDown.js#L9-L12

Reply:
@KsenijaZivkovic and I talked about this and we agreed to create another map, the `COIN_TYPE_TO_META` maps coinType to contractName, where in this use case we need the reversed mapping, sticking with `COIN_TYPE_TO_META` will require looping through the list of all coinTypes for each of the contract name we get, I think it's worthwhile to trade space for time for this instance.

Reply:
i see 👍 . In this case support a new coin type will require changes at multiple places, maybe worth to add comment or something to help remind ppl these "connected" mappings. If we can have unit test to cover this mapping would be very helpful to serve as a safe guard.

Reply:
It would be ideal to add an API integration that would know how to map this correctly, @jacksonConrad 
 sent me this https://docs.flowgraph.co/examples/contract/token/
 Maybe for now to leave it hardcoded (add comment) and plan the integration for the future